crmprtd/align.py

```diff
@@ -8,12 +8,237 @@
+                            History.id == id).first()
+                    except Exception as e:
+                        log.warning('An error occured while accesing the db',
+                                    extra={'exception': e})
+
+                    dist = haversine_formula(obs_tuple.lat,
+                                             obs_tuple.lon,
+                                             float(lat),
+                                             float(lon))
+
+                    # best case, we find exact match
+                    if dist == 0:
+                        log.debug('Exact match found')
+                        hid = id
+                        break
+
+                    # find closest
+                    else:
+                        if dist < closest_pos:
+                            hid = id
+                            closest_pos = dist
+
+            if not hid:
+                log.debug('Observation contains lat/lon but does not match '
+                          'any entries in db')
+                hid = create_station_and_history_entry(sesh, obs_tuple)
+        else:
+            log.debug('No lat/lon, checking for \"active\" stations')
+            for id in q.all():
+                q = sesh.query(History.id).filter(
+                    and_(History.id == id,
+                         History.sdate is not None,
+                         History.edate is None))
+
+                if q.count() > 0:
+                    log.debug('Matched hid using sdate/edate')
+                    hid, = q.first()
+                    break
+
+        if not hid:
+            log.warning('Unable to match station',
+                        extra={'station_id': obs_tuple.station_id,
+                               'network_name': obs_tuple.network_name})
+            return
+
+    # thing
+    log.debug('Check time')
+    if obs_tuple.time is None:
+        log.error('Observation cannot be used without time')
+        return
+    log.debug('Observation has time')
+    time = obs_tuple.time
+
+    log.debug('Check data')
+    if obs_tuple.val is None:
+        log.error('Observation cannot be used without value')
+        return
+    log.debug('Observation has value')
+    datum = obs_tuple.val
+
+    log.debug('Check if variable name exists in database')
+    q = sesh.query(Variable.id).join(Network).filter(and_(
+        Network.name == obs_tuple.network_name,
+        Variable.name == obs_tuple.variable_name))
+
+    if q.count() < 1:
+        log.warning('No matching variable found', extra={
+                    'vars_id': obs_tuple.variable_name})
+        return
+
+    vars_id, = q.first()
+    log.debug('Observation variable matches', extra={'vars_id': vars_id})
+
+    log.debug('Check unit')
+    if obs_tuple.unit is None:
+        log.debug('No unit given, checking if table contains unit')
+        unit, = sesh.query(Variable.unit).join(Network).filter(and_(
```

Comment:
If you simply query for `Variable` above (as I described early on in this review), then you can simply use the expression `variable.unit` here instead of explicitly writing another query.